code/modules/reagents/reagent_containers.dm

```diff
@@ -265,6 +265,19 @@
 			if(C.has_udder)
 				return
 
+		if(istype(target, /mob/living/simple_animal/carp))
+			var/mob/living/simple_animal/spaceCarp = target
+			if(spaceCarp.has_toxingland)
+				return
+
+		if(istype(target, /mob/living/simple_animal/carp/fluff/ginny))
+			return
+
+		if(istype(target, /mob/living/simple_animal/hostile/carp))
+			var/mob/living/simple_animal/HostileCarp = target
+			if(HostileCarp.has_toxingland)
+				return
```

Comment:
```suggestion			
		if(is_type_in_list(target, list(/mob/living/simple_animal/carp, /mob/living/simple_animal/hostile/carp)))
			var/mob/living/simple_animal/carp_being_fed= target
			if(istype(carp_being_fed, /mob/living/simple_animal/carp/fluff/ginny) || carp_being_fed.has_toxingland)
				return
```
Not sure why you're checking this for feeding though, why does it matter that it has a toxingland on if it can be fed or not? And why can't you feed ginny?

Reply:
I was running into a weird issue where using a beaker on a carp (Ginny included) would attempt to feed the carp the contents of the beaker. It's been some time since I made the changes, and I can't quite remember what I did, but I recall copying the code that prevented you from forcing cows to drink their own milk if you tried to milk them consecutively.

Reply:
I think I've got it working. Thanks for helping with the code.